src/mds/OpenFileTable.cc

```diff
@@ -465,8 +465,10 @@ void OpenFileTable::commit(MDSContext *c, uint64_t log_seq, int op_prio)
 	ceph_assert(it.second == DIRTY_NEW);
 	// find omap object to store the key
 	for (unsigned i = first_free_idx; i < omap_num_objs; i++) {
-	  if (omap_num_items[i] < MAX_ITEMS_PER_OBJ)
+	  if (omap_num_items[i] < MAX_ITEMS_PER_OBJ) {
 	    omap_idx = i;
+	    break;
+	  }
```

Comment:
I don't see how this will fix the bug. The current behavior of this loop is that the last free object is chosen. You're flipping the behavior by picking the first available one (a reasonable change by itself). However, how does that avoid the assertion below?

The commit message could be improved...

Reply:
For example:

- object 0  reach max item count. mds create object 1.
- mds delete items from object 0
- mds serach for free object, 

because last_free_object is assigned to first_free_idx.  when mds searches free object in next around, it will skip preceding free object.

Reply:
Yes, I see that but how does it result in the last free object going over MAX_OBJECTS, thus hitting the assertion?

Reply:
The old behavior is:

- If the last object contains at least on item, it will be chosen for new item.
- when the last object is fully, new object is allocated (new last object).

For above reasons, the object count may keep increasing